hooks/browser_integration.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import os
+import hashlib
+import json
+import fnmatch
+import datetime
+
+import sgtk
+
+class BrowserIntegration(sgtk.Hook):
+    def filter_engine_commands(self, commands):
+        """
+        Filters out engine commands that are not desired in the web app menus.
+
+        :param list commands: The list of engine commands to filter.
```

Comment:
i think for each of these hook methods, it would be great if we explained in the docstring how a user could override them and why this could be useful. Some simple use case to help illustrate, not what they currently do, but why we expose them as hooks and why they are useful.